library/src/main/java/com/getstream/sdk/chat/rest/core/providers/StreamWebSocketServiceProvider.java

```diff
@@ -32,13 +32,30 @@ public StreamWebSocketServiceProvider(ApiClientOptions options, String apiKey) {
     }
 
     @Override
-    public WebSocketService provideWebSocketService(User user, String userToken, WSResponseHandler listener) throws UnsupportedEncodingException {
+    public WebSocketService provideWebSocketService(User user, @Nullable String userToken, WSResponseHandler listener) throws UnsupportedEncodingException {
         String wsUrl = getWsUrl(userToken, user);
         Log.d(TAG, "WebSocket URL : " + wsUrl);
         return new StreamWebSocketService(wsUrl, listener);
     }
 
-    @NotNull
+    public String getWsUrl(@Nullable String userToken, User user) throws UnsupportedEncodingException {
+        String json = buildUserDetailJSON(user);
+
+        try {
+            json = URLEncoder.encode(json, StandardCharsets.UTF_8.toString());
+            String baseWsUrl = apiClientOptions.getWssURL() + "connect?json=" + json + "&api_key=" + apiKey;
```

Comment:
There is too much code inside the try/except; we should only wrap the code that can throw the exception 

 ```java
        try {
            json = URLEncoder.encode(json, StandardCharsets.UTF_8.toString());
        } catch (Throwable throwable) {
            throwable.printStackTrace();
            throw new UnsupportedEncodingException("Unable to encode user details json: " + json);
        }
       // ...
```